hugo/README.md

```diff
@@ -0,0 +1,201 @@
+  
+- You can use relative paths or just the filename. 
+- Paths without a leading `/` are first resolved relative to the current page, then to the remainder of the site.
+- Anchors are supported.
+
+For example:
+
+```md
+To install NGINX Controller, refer to the [installation instructions]({{< ref "install" >}}).
+```
+
+### Add images
+
+You can use the `img` [shortcode](#shortcodes) to insert images into your documentation.
+
+1. Add the image to the static/img directory, or to the same directory as the doc you want to use it in.
+   DO NOT include a forward slash at the beginning of the file path. This will break the image when it's rendered.
+   See the docs for the [Hugo relURL Function](https://gohugo.io/functions/relurl/#input-begins-with-a-slash) to learn more.
+
+1. Add the img shortcode:
+
+    {{< img src="<img-file.png>" >}}
+ 
+> Note: The shortcode accepts all of the same parameters as the [Hugo figure shortcode](https://gohugo.io/content-management/shortcodes/#figure). 
+
+### Use Hugo shortcodes
+You can use Hugo [shortcodes](https://gohugo.io/content-management/shortcodes) to do things like format callouts, add images, and reuse content across different docs. 
+
+For example, to use the note callout:
+
+```md
+{{< note >}}Provide the text of the note here. {{< /note >}}
+``` 
+
+The callout shortcodes also support multi-line blocks:
+
+```md
+{{< caution >}}
+You should probably never do this specific thing in a production environment. If you do, and things break, don't say we didn't warn you.
+{{< /caution >}}
+```
+
+Supported callouts:
+- caution
+- important
+- note
+- see-also
+- tip
+- warning
+
+A few more useful shortcodes:
+
+- collapse: makes a section collapsible
+- table: adds scrollbars to wide tables when viewed in small browser windows or mobile browsers
+- fa: inserts a Font Awesome icon
+- include: include the content of a file in another file (requires the included file to be in the /includes directory)
+- link: makes it possible to link to a static file and prepend the path with the Hugo baseUrl
+- openapi: loads an OpenAPI spec and renders as HTML using ReDoc
+- raw-html: makes it possible to include a block of raw HTML
+- readfile: includes the content of another file in the current file; useful for adding code examples
+
+## How to build docs locally
+
+To view the docs in a browser, run the Hugo server. This will reload the docs automatically so you can view updates as you work.
+
+> Note: The docs use build environments to control the baseURL that will be used for things like internal references and resource (CSS and JS) loading. 
+> You can view the config for each environment in the [config](./config) directory of this repo.
+When running the Hugo server, you can specify the environment and baseURL if desired, but it's not necessary.
+
+For example:
+
+```
+hugo server
+```
+
+```
+hugo server -e development -b "http://127.0.0.1/nginx-gateway-fabric/"
```

Comment:
will anyone be able o run it or it requires access to private (F5) repos?

Reply:
the only dependency is the hugo-theme which is in a public repo:
https://github.com/nginxinc/nginx-hugo-theme